diff --git a/node_modules/eleventy-plugin-i18n-gettext/dist/i18n.js b/node_modules/eleventy-plugin-i18n-gettext/dist/i18n.js
index 78a29dc..e700cf8 100644
--- a/node_modules/eleventy-plugin-i18n-gettext/dist/i18n.js
+++ b/node_modules/eleventy-plugin-i18n-gettext/dist/i18n.js
@@ -75,12 +75,12 @@ class i18n {
         return obj;
     }
     normalizePath(path) {
-        if (this.pathPrefix === undefined) {
-            // Works when pathPrefix is configured in .eleventy.js file
-            // Does not work when pathPrefix is given with commandline `--pathprefix=eleventy-base-blog`
-            const projectConfig = require('@11ty/eleventy/src/Config').getConfig();
-            this.pathPrefix = projectConfig.pathPrefix;
-        }
+        // if (this.pathPrefix === undefined) {
+        //     // Works when pathPrefix is configured in .eleventy.js file
+        //     // Does not work when pathPrefix is given with commandline `--pathprefix=eleventy-base-blog`
+        //     const projectConfig = require('@11ty/eleventy/src/Config').getConfig();
+        //     this.pathPrefix = projectConfig.pathPrefix;
+        // }
         if (this.pathPrefix !== '/') {
             return path.replace(this.pathPrefix, '/');
         }
